prow/config/branch_protection.go

```diff
@@ -142,19 +146,24 @@ func jobRequirements(jobs []Presubmit, branch string, after bool) []string {
 		if !after && !j.AlwaysRun && j.RunIfChanged == "" {
 			continue // No
 		}
-		if !j.SkipReport && !j.Optional { // This job needs a context
+		if j.ContextRequired() { // This job needs a context
 			required = append(required, j.Context)
+		} else {
+			optional = append(optional, j.Context)
 		}
 		// Check which children require contexts
-		required = append(required, jobRequirements(j.RunAfterSuccess, branch, true)...)
+		r, o := jobRequirements(j.RunAfterSuccess, branch, true)
+		required = append(required, r...)
+		optional = append(optional, o...)
 	}
-	return required
+	return
```

Comment:
-1 to named returns

Reply:
Agreed on not using named returns

Reply:
May I ask why ?

Reply:
I find almost universally they are harder to reason about and read through. They are almost nonexistant in the codebase as it is. Stats:

```
$ grep -rh func prow/ | grep -Po "(?<=\) \()[^\(\)]+(?=\) \{)" | tr ',' '\n' | awk '{print $2 }' | grep -v "^$" | wc -l 
19
$ grep -rh func prow/ | wc -l 
2084
```

Reply:
I actually discovered them in k8s codebase :) but right it may be confusing ! fixed.